lib/literal/railtie.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class Literal::Railtie < Rails::Railtie
+	initializer "literal.register_literal_enum_type" do
+		ActiveRecord::Type.register(:literal_enum) do |name, type:|
+			Literal::Rails::EnumType.new(type)
+		end
+
+		ActiveModel::Type.register(:literal_enum) do |name, type:|
+			Literal::Rails::EnumType.new(type)
+		end
```

Comment:
This gives us the interface

```ruby
attribute :status, :literal_enum, type: Status
```

It’s not perfect but I think it’s better than any of the other options.